examples/inverse/plot_evoked_ers_source_power.py

```diff
@@ -0,0 +1,144 @@
+epochs = mne.Epochs(raw, events, event_id=1, tmin=-1.5, tmax=2, picks=picks,
+                    preload=True)
+
+# Read forward operator and point to freesurfer subject directory
+fname_fwd = op.join(data_path, 'derivatives', 'sub-{}'.format(subject),
+                    'sub-{}_task-{}-fwd.fif'.format(subject, task))
+subjects_dir = op.join(data_path, 'derivatives', 'freesurfer', 'subjects')
+
+fwd = mne.read_forward_solution(fname_fwd)
+
+# fix the subject in fwd['src']
+for src in fwd['src']:
+    src['subject_his_id'] = subject
+
+###############################################################################
+# now we can compute some source estimates using different methods.
+#
+# define the active and baseline windows in seconds
+# ERS activity starts at 0.5 seconds after stimulus onset
+active_win = (0.5, 1.5)
+baseline_win = (-1, 0)
+
+
+###############################################################################
+# generate a dics source estimate - see :ref:`ex-inverse-source-power` for
+# more information
+def _gen_dics(active_win, baseline_win, epochs):
+    freqs = np.logspace(np.log10(12), np.log10(30), 9)
+    csd = csd_morlet(epochs, freqs, tmin=-1, tmax=1.5, decim=20)
+    csd_baseline = csd_morlet(epochs, freqs, tmin=baseline_win[0],
+                              tmax=baseline_win[1], decim=20)
+    csd_ers = csd_morlet(epochs, freqs, tmin=active_win[0], tmax=active_win[1],
+                         decim=20)
+    filters = make_dics(epochs.info, fwd, csd.mean(), pick_ori='max-power')
+    baseline_source_power, freqs = apply_dics_csd(csd_baseline.mean(), filters)
+    beta_source_power, freqs = apply_dics_csd(csd_ers.mean(), filters)
+    stc = beta_source_power / baseline_source_power
+    return stc
+
+
+# generate lcmv source estimate
+def _gen_lcmv(active_cov, baseline_cov, common_cov):
+    filters = make_lcmv(epochs.info, fwd, common_cov, reg=0.05,
+                        noise_cov=None, pick_ori='max-power',
+                        weight_norm='nai', rank=None)
+    stc_base = apply_lcmv_cov(baseline_cov, filters)
+    stc_act = apply_lcmv_cov(active_cov, filters)
+    stc_act /= stc_base
+    return stc_act
+
+
+# generate mne/dSPM source estimate
+def _gen_mne(active_cov, baseline_cov, common_cov, fwd, info, method):
+    inverse_operator = make_inverse_operator(info, fwd, common_cov,
+                                             loose=0.2, depth=0.8)
+
+    stc_act = apply_inverse_cov(
+        active_cov, info, 1, inverse_operator,
+        method=method, pick_ori=None,
+        lambda2=1. / 9.,
+        verbose=True, dB=False)
+
+    stc_base = apply_inverse_cov(
+        baseline_cov, info, 1, inverse_operator,
+        method=method, pick_ori=None,
+        lambda2=1. / 9.,
+        verbose=True, dB=False)
+    stc_act /= stc_base
+    return stc_act
+
+
+###############################################################################
+# compute covariances needed for the lcmv and MNE methods.
```

Comment:
```suggestion
# Compute covariance matrices needed for the LCMV and MNE methods.
```

Also, why is this done here and not above the `_gen_*` functions? The `_gen_*` functions need the covariance matrices and it is weird to me "story" wise to compute them later.